src/commonMain/kotlin/fr/acinq/lightning/channel/WaitForFundingSigned.kt

```diff
@@ -2,100 +2,126 @@ package fr.acinq.lightning.channel
 
 import fr.acinq.bitcoin.BlockHeader
 import fr.acinq.bitcoin.ByteVector32
-import fr.acinq.bitcoin.Satoshi
-import fr.acinq.bitcoin.Transaction
+import fr.acinq.bitcoin.PublicKey
+import fr.acinq.bitcoin.crypto.Pack
 import fr.acinq.lightning.Feature
-import fr.acinq.lightning.Lightning
+import fr.acinq.lightning.MilliSatoshi
+import fr.acinq.lightning.ShortChannelId
 import fr.acinq.lightning.blockchain.BITCOIN_FUNDING_DEPTHOK
 import fr.acinq.lightning.blockchain.BITCOIN_FUNDING_SPENT
 import fr.acinq.lightning.blockchain.WatchConfirmed
 import fr.acinq.lightning.blockchain.WatchSpent
+import fr.acinq.lightning.blockchain.electrum.WalletState
 import fr.acinq.lightning.blockchain.fee.OnChainFeerates
-import fr.acinq.lightning.crypto.ShaChain
-import fr.acinq.lightning.transactions.CommitmentSpec
-import fr.acinq.lightning.transactions.Transactions
-import fr.acinq.lightning.utils.Either
-import fr.acinq.lightning.utils.Try
-import fr.acinq.lightning.wire.Error
-import fr.acinq.lightning.wire.FundingCreated
-import fr.acinq.lightning.wire.FundingSigned
+import fr.acinq.lightning.wire.*
+import kotlin.math.absoluteValue
 
 data class WaitForFundingSigned(
```

Comment:
This state handles the exchange of `commit_sig` (which makes me realize that I should add comments to each state class to explain what part of the protocol they're supposed to handle).

We could do it in `WaitForFundingCreated`, but it becomes a bit messier because we'd have to keep more variable state inside, I didn't like it...